docs/gen-ai/llm-spans.md

```diff
@@ -36,49 +50,356 @@ Instrumentations that support it, MUST offer the ability to turn off capture of
 
-**[3]:** If available. The name of the LLM serving a response. If the LLM is supplied by a vendor, then the value must be the exact name of the model actually used. If the LLM is a fine-tuned custom model, the value should have a more specific name than the base model that's been fine-tuned.
+**[3]:** If there is more than one finish reason in the response, the last one should be reported.
+
+**[4]:** If available. The name of the LLM serving a response. If the LLM is supplied by a vendor, then the value must be the exact name of the model actually used. If the LLM is a fine-tuned custom model, the value should have a more specific name than the base model that's been fine-tuned.
+
+`gen_ai.system` has the following list of well-known values. If one of them applies, then the respective value MUST be used; otherwise, a custom value MAY be used.
+
+| Value  | Description | Stability |
+|---|---|---|
+| `openai` | OpenAI | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 <!-- endsemconv -->
 
 ## Events
 
-In the lifetime of an LLM span, an event for prompts sent and completions received MAY be created, depending on the configuration of the instrumentation.
+In the lifetime of an GenAI span, an event for each message application sends to GenAI and receives in response from it MAY be created, depending on the configuration of the instrumentation.
+The generic events applicable to multiple GenAI models follow `gen_ai.{type}.message` naming pattern.
+GenAI vendor-specific instrumentations SHOULD follow `gen_ai.{gen_ai.system}.{type}.message` pattern to record events that apply to their system only.
+
+It's RECOMMENDED to use [Event API](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/event-api.md) to record GenAI events once it's implemented in corresponding language.
+
+If, however Event API is not supported yet, events SHOULD be recorded as span events.
+
+TODO: There [will be](https://github.com/open-telemetry/semantic-conventions/pull/954) a standard mapping between Span Events and Events. Add link once merged.
 
-<!-- semconv gen_ai.content.prompt -->
-The event name MUST be `gen_ai.content.prompt`.
+The event payload describes message content sent to or received from GenAI and depends on specific messages described in the following sections.
+
+Instrumentations for individual GenAI systems MAY add system specific fields into corresponding events payload.
+It's RECOMMENDED to document them in system-specific extensions.
+Telemetry consumers SHOULD expect to receive unknown payload fields.
+
+### System message
```

Comment:
Are we not going to support legacy completion APIs that accept single prompts?

Reply:
Legacy completion fits into the user message quite well.